### PR TITLE
Laboratory work #2, Alina Batrasova - 23FPL1

### PR DESCRIPTION
**Pull request by AlinaBatrasova**

### CODE REVIEW

File: lab_2_retrieval_w_bm25/main.py [hunk 1]
```diff
@@ -47,7 +65,22 @@ def build_vocabulary(documents: list[list[str]]) -> list[str] | None:
...
+    for doc in documents:
+        if not all(isinstance(word, str) for word in doc):
+            return None
+
+    unique = []
+
+    for doc in documents:
+        for word in doc:
+            if word not in unique:
+                unique.append(word)
```
Comment (Vasilisa-Blyudova) [inline lab_2_retrieval_w_bm25/main.py:81]: use list comprehensions

File: lab_2_retrieval_w_bm25/main.py [hunk 2]
```diff
@@ -62,6 +95,22 @@ def calculate_tf(vocab: list[str], document_tokens: list[str]) -> dict[str, floa
...
+
+    tf = {}
+    for stopword in document_tokens:
+        if stopword not in vocab:
+            tf[stopword] = document_tokens.count(stopword) / len(document_tokens)
+    for word in vocab:
+        tf[word] = 0.0
+        if word in document_tokens or tf[word] != 0.0:
+            tf[word] = document_tokens.count(word) / len(document_tokens)
+            continue
```
Comment (Vasilisa-Blyudova) [inline lab_2_retrieval_w_bm25/main.py:111]: why is the continue operator needed here?

File: seminars/practice_3_lists.py [hunk 3]
```diff
@@ -60,13 +60,15 @@ def count_evens(nums: list) -> int:
     """
     Return the number of even ints in the given array.
     """
-    # student realization goes here
+    # counter_even = 0
```
Comment (Vasilisa-Blyudova) [inline seminars/practice_3_lists.py:63]: rm all changes from seminars

**Review by Vasilisa-Blyudova — COMMENTED** [review]
Good job! Fix the tests and comments and raise the score to 8!

**Comment by Vasilisa-Blyudova** [comment]
class Discipline :
    """
    A class to represent a learning discipline studied in HSE
    """
    def __init__(self, name_subject, name_teacher, credits, hours, tasks : list, stud_grades : dict):
        self.name_subject = name_subject
        self.name_teacher = name_teacher
        self.credits = credits
        self.hours = hours
        self.tasks = tasks
        self.stud_grades = stud_grades

    def introduce(self):
        print(f'Your new discuipline is {self.name_subject}, teacher is {self.name_teacher}. Additional info: credits = {self.credits}, learning hours = {self.hours}, tasks = {self.tasks}')

    def get_task(self):
        for task in self.tasks:
            print(task)

    def get_best_stud(self):
        print(sorted(self.stud_grades, key = lambda x: x ))
        sorted_dict = sorted(self.stud_grades.values(), key = lambda x: -x )
        #best_stud = self.stud_grad
        print(sorted_dict)

vvl = Discipline('Введение в лингвистику', 'Маргарита Андреевна', 100, 1100, ['реферирование', "самостоятельная 1", "самостоятельная 2", "экзамен"], {'aaaa' : 2, 'Иванов' : 1, "Фокина" : 13} )
vvl.introduce()
vvl.get_task()
vvl.get_best_stud()

**Comment by Vasilisa-Blyudova** [comment]
Unfortunately, no progress since the last review!

**Comment by Vasilisa-Blyudova** [comment]
Unfortunately, I don't see any progress in your PR since the last review.

File: seminars/practice_3_lists.py (hunk 3 above)
Comment (AlinaBatrasova) [reply seminars/practice_3_lists.py:63]: done

File: lab_2_retrieval_w_bm25/main.py (hunk 2 above)
Comment (AlinaBatrasova) [reply lab_2_retrieval_w_bm25/main.py:111]: it's not needed, yes... fixed

File: lab_2_retrieval_w_bm25/main.py (hunk 1 above)
Comment (AlinaBatrasova) [reply lab_2_retrieval_w_bm25/main.py:82]: i don'r really get how to write it :(

**Pull request closed without merging**